src/ethereum/prague/requests.py

```diff
@@ -30,178 +21,53 @@
     receipt: Union[Bytes, Receipt],
-) -> Tuple[Bytes, ...]:
+) -> Bytes:
     """
     Parse deposit requests from a receipt.
     """
-    deposit_requests: Tuple[Bytes, ...] = ()
+    deposit_requests: Bytes = b""
     decoded_receipt = decode_receipt(receipt)
     for log in decoded_receipt.logs:
         if log.address == DEPOSIT_CONTRACT_ADDRESS:
-            deposit_request = parse_deposit_data(log.data)
-            deposit_requests += (encode_request(deposit_request),)
+            request = parse_deposit_data(log.data)
+            deposit_requests += request
 
     return deposit_requests
 
 
-def parse_withdrawal_data(data: Bytes) -> WithdrawalRequest:
+def compute_requests_hash(requests: List[Bytes]) -> Bytes:
     """
-    Parses Withdrawal Request from the data.
-    """
-    assert len(data) == WITHDRAWAL_REQUEST_LENGTH
-    return WithdrawalRequest(
-        source_address=Address(data[:20]),
-        validator_public_key=Bytes48(data[20:68]),
-        amount=U64.from_be_bytes(data[68:76]),
-    )
+    Get the hash of the requests.
 
+    Parameters
+    ----------
+    requests : Bytes
+        The requests to hash.
 
-def parse_withdrawal_requests_from_system_tx(
-    evm_call_output: Bytes,
-) -> Tuple[Bytes, ...]:
-    """
-    Parse withdrawal requests from the system transaction output.
+    Returns
+    -------
+    requests_hash : Bytes
+        The hash of the requests.
     """
-    count_withdrawal_requests = (
-        len(evm_call_output) // WITHDRAWAL_REQUEST_LENGTH
-    )
-
-    withdrawal_requests: Tuple[Bytes, ...] = ()
-    for i in range(count_withdrawal_requests):
-        start = i * WITHDRAWAL_REQUEST_LENGTH
-        withdrawal_request = parse_withdrawal_data(
-            evm_call_output[start : start + WITHDRAWAL_REQUEST_LENGTH]
-        )
-        withdrawal_requests += (encode_request(withdrawal_request),)
-
-    return withdrawal_requests
-
-
-def parse_consolidation_data(data: Bytes) -> ConsolidationRequest:
-    """
-    Parses Consolidation Request from the data.
-    """
-    assert len(data) == CONSOLIDATION_REQUEST_LENGTH
-    return ConsolidationRequest(
-        source_address=Address(data[:20]),
-        source_public_key=Bytes48(data[20:68]),
-        target_public_key=Bytes48(data[68:116]),
-    )
-
-
-def parse_consolidation_requests_from_system_tx(
-    evm_call_output: Bytes,
-) -> Tuple[Bytes, ...]:
-    """
-    Parse consolidation requests from the system transaction output.
-    """
-    count_consolidation_requests = (
-        len(evm_call_output) // CONSOLIDATION_REQUEST_LENGTH
-    )
-
-    consolidation_requests: Tuple[Bytes, ...] = ()
-    for i in range(count_consolidation_requests):
-        start = i * CONSOLIDATION_REQUEST_LENGTH
-        consolidation_request = parse_consolidation_data(
-            evm_call_output[start : start + CONSOLIDATION_REQUEST_LENGTH]
-        )
-        consolidation_requests += (encode_request(consolidation_request),)
+    m = sha256()
```

Comment:
Weird. We should make a note somewhere (probably in `ethereum.crypto.hash`) of exactly what algorithm this is. I believe it's SHA2-256, right?

Reply:
Added a note to the function doc string